iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/partition/SeriesPartitionTable.java

```diff
@@ -245,10 +246,19 @@ public TConsensusGroupId getLastConsensusGroupId() {
    * @param TTL The Time To Live
    * @param currentTimeSlot The current TimeSlot
    */
-  public void autoCleanPartitionTable(long TTL, TTimePartitionSlot currentTimeSlot) {
+  public List<TTimePartitionSlot> autoCleanPartitionTable(
+      long TTL, TTimePartitionSlot currentTimeSlot) {
+    List<TTimePartitionSlot> removedTimePartitions = new ArrayList<>();
```

Comment:
Using an iterator for a single pass?

Reply:
Could u plz teach me how to spell this proficient Java code? I am certainly not good at this :)

Reply:
```
List<TTimePartitionSlot> removedTimePartitions = new ArrayList<>();
Iterator<Map.Entry<TTimePartitionSlot, List<String>>> iterator = seriesPartitionMap.entrySet().iterator();

while (iterator.hasNext()) {
    Map.Entry<TTimePartitionSlot, List<String>> entry = iterator.next();
    TTimePartitionSlot timePartitionSlot = entry.getKey();
    if (timePartitionSlot.getStartTime() + TTL < currentTimeSlot.getStartTime()) {
        removedTimePartitions.add(timePartitionSlot);
        iterator.remove();
    }
}
return removedTimePartitions;
```

Reply:
These elegant codes do simplify the whole function significantly! Thanks for guding!